api/routes/mongoose.ts

```diff
@@ -0,0 +1,23 @@
+import { FastifyInstance } from 'fastify';
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export async function mongooseRoute(fastify: FastifyInstance) {
+  fastify.get('/test-mongoose', async (_request, _reply) => {
+    const userModel = fastify.mongoose.user;
+    // findOne's type parameter needs to be specified if you want the user type
+    // to account for the projection. However, it can be any type:
+    const user = await userModel.findOne<{
```

Comment:
```suggestion
    const User = fastify.User;
    // findOne's type parameter needs to be specified if you want the user type
    // to account for the projection. However, it can be any type:
    const user = await User.findOne<{
```